docs/types/mapping-types.rst

```diff
@@ -52,6 +58,54 @@ each ``_KeyType``, recursively. For example with a mapping:
         }
     }
 
+The example below is a simplified version of an `ERC20 token <https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol>`_.
+``_allowances`` is an example of a mapping type inside another mapping type.
+The example below uses ``_allowances`` to record the amount of a transaction and the two addresses it was between.
+
+::
+
+    pragma solidity >=0.4.0 <0.7.0;
+
+    contract MappingExample {
+
+        mapping (address => uint256) private _balances;
+        mapping (address => mapping (address => uint256)) private _allowances;
+
+        event Transfer(address indexed from, address indexed to, uint256 value);
+        event Approval(address indexed owner, address indexed spender, uint256 value);
+
+        function allowance(address owner, address spender) public view returns (uint256) {
+            return _allowances[owner][spender];
+        }
+
+        function transferFrom(address sender, address recipient, uint256 amount) public returns (bool) {
+            _transfer(sender, recipient, amount);
+            _approve(sender, msg.sender, amount);
```

Comment:
Why do you call approve here? It should rather check that the allowance is large enough.

Reply:
@chriseth This came from the Zeppelin code…

Reply:
But it does something else now.